### PR TITLE
Add Lift generator

### PR DESCRIPTION
**Pull request by andrei-sa**

### CODE REVIEW

File: Tests/Source/LiftTester.cpp [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+          return std::make_tuple(v1, v2);
+        }, r4, r5);
+      }();
+      auto result = generate(generator);
+      REQUIRE(std::get<0>(result) >= 5);
+      REQUIRE(std::get<0>(result) <= std::get<1>(result));
+      REQUIRE(std::get<1>(result) <= 50);
+    }
+  }
+}
```
Comment (kranar) [inline Tests/Source/LiftTester.cpp]: End with a new line.

File: Include/Rover/Lift.hpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#ifndef ROVER_LIFT_HPP
+#define ROVER_LIFT_HPP
+#include <tuple>
+#include "Rover/Evaluator.hpp"
```
Comment (kranar) [inline Include/Rover/Lift.hpp:5]: Include ```<type_traits>``` for ```invoke_result_t``` and ```decay_t``` etc...

File: Include/Rover/Lift.hpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  template<typename FunctionFwd, typename... GeneratorsFwd>
+  Lift(FunctionFwd&&, GeneratorsFwd&&...) -> Lift<std::decay_t<FunctionFwd>,
+      std::decay_t<GeneratorsFwd>...>;
+
+  template<typename F, typename... Generators>
+  template<typename FunctionFwd, typename... GeneratorsFwd>
+  constexpr Lift<F, Generators...>::Lift(FunctionFwd&& func,
+      GeneratorsFwd&&... generators)
+      : m_func(std::forward<FunctionFwd>(func)),
+        m_generators(std::forward_as_tuple(generators...)) {}
```
Comment (kranar) [inline Include/Rover/Lift.hpp]: ```forward_as_tuple``` is unlikely what we want here as we end up losing the value category of the ```generator``` arguments. ```m_generators(std::forward<GeneratorsFwd>(generators)...)``` will do the trick though.

**Review by kranar — CHANGES_REQUESTED** [review]
This is excellent, it's exactly how it should work.

File: Include/Rover/Lift.hpp (hunk 3 above)
Comment (andrei-sa) [reply Include/Rover/Lift.hpp]: No idea why I put it this way.

**Pull request merged**